File: 2022/Sources/advent2022/Day1.swift
struct Day1 : Day {
    let sampleInput = [
        "1000", "2000", "3000", "", "4000", "", "5000", "6000", "", "7000", "8000", "9000", "", "10000"
    ]
    func readElves (_ lines :[String]) throws -> [Int] { lines.reduce([0], { es, elem in
        elem == "" ? es + [0] : es.prefix(es.count-1) + [es.last! + Int(elem)!]
    })}

    func part1 () throws -> String { String(try readElves(try readInput(1)).max()!) }

    func part2 () throws -> String {
        var elves = try readElves(try readInput(1))
        elves.sort(by: >)
        return String(elves.prefix(3).reduce(0, +))
     }
}

File: 2022/scala/src/main/scala/Day1.scala
object Day1 extends Day(1):

  def readElves (lines :Seq[String]) :Seq[Int] = lines.foldLeft(Seq(0)) {
    (es, elem) => if elem == "" then es :+ 0 else es.dropRight(1) :+ es.last + elem.toInt }

  override def answer1 (input :Seq[String]) = readElves(input).max
  override def answer2 (input :Seq[String]) = readElves(input).sortWith(_ > _).take(3).sum
